br/pkg/lightning/backend/kv/base.go

```diff
@@ -93,6 +94,13 @@ func (row RowArrayMarshaller) MarshalLogArray(encoder zapcore.ArrayEncoder) erro
 			if err != nil {
 				return err
 			}
+			if len(str) > 512*1024 {
+				str = str[0:1024]
+			}
+			totalLength += len(str)
```

Comment:
not important: we can move it to one level outside, so for all `switch kind` we will accumulate the length.